### PR TITLE
Adds test to ensure that multiple federators that share at least one …

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds test to ensure that multiple federators that share at least one of the same keys are considered the same

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2620,6 +2620,80 @@ void voteFederationChange_addFederatorMultiKeyWithDifferentMstKey_returnsSuccess
...
+
+        @ParameterizedTest
+        @MethodSource("federatorsWithDifferentKeysButOneParametersProvider")
+        void voteFederationChange_add2FederatorsWithWithDifferentKeysButOne_returnsFederatorAlreadyPresentResponseCode(
+            BtcECKey federator1BtcKey,
+            ECKey federator1RskKey,
+            ECKey federator1MstKey,
+            BtcECKey federator2BtcKey,
+            ECKey federator2RskKey,
+            ECKey federator2MstKey
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2651]:
```suggestion
            FederationMember ... federationMembers
```

Instead use `FederationMember` class.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -2620,6 +2620,80 @@ void voteFederationChange_addFederatorMultiKeyWithDifferentMstKey_returnsSuccess
...
+
+            // Voting with m of n authorizers to create the pending federation
+            federationSupport.voteFederationChange(firstAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(secondAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            ABICallSpec addMultiKeyAbiCallSpec1 = new ABICallSpec(FederationChangeFunction.ADD_MULTI.getKey(), new byte[][]{
+                federator1BtcKey.getPubKey(),
+                federator1RskKey.getPubKey(),
+                federator1MstKey.getPubKey(),
+            });
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2667]: You could create a helper method to create the byte[][] with the keys. Wdyt?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2651]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2667]: Updated.

**Review by nathanieliov — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**